src/openexr.imageio/exroutput.cpp

```diff
@@ -290,6 +290,197 @@ set_exr_threads()
+    0.0f,     1.0f,     // green
+    0.0001f,  -0.077f,  // blue
+    0.32168f, 0.33767f  // white
+};
+
+static const std::string ACES_AP0_colorInteropId = "lin_ap0_scene";
+
+
+bool
+is_spec_aces_container_channels_only(const OIIO::ImageSpec& spec)
+{
+    // Note: this is constructing and comparing sets, so that channel order
+    // doesn't matter.
+
+    // Allowed channel sets
+    static const std::vector<std::set<std::string>> allowed_sets
+        = { { "B", "G", "R" },
+            { "A", "B", "G", "R" },
+            { "B", "G", "R", "left.B", "left.G", "left.R" },
+            { "A", "B", "G", "R", "left.A", "left.B", "left.G", "left.R" } };
+
+    // Gather channel set from spec
+    std::set<std::string> channels(spec.channelnames.begin(),
+                                   spec.channelnames.end());
+
+    // Compare to allowed sets (unordered)
+    for (const auto& allowed : allowed_sets) {
+        if (channels == allowed) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+
+
+bool
+is_aces_container_attributes_non_empty(const OIIO::ImageSpec& spec,
+                                       std::string& non_compliant_attr)
+{
+    // attributes in this list should NOT be empty if they exist
+    static const std::vector<std::string> nonEmptyAttribs = {
+        "cameraFirmwareVersion",
+        "cameraIdentifier",
+        "cameraLabel",
+        "cameraMake",
+        "cameraModel",
+        "cameraSerialNumber",
+        "comments",
+        "creator",
+        "lensAttributes",
+        "lensFirmwareVersion",
+        "lensMake",
+        "lensModel",
+        "lensSerialNumber",
+        "owner",
+        "recorderFirmwareVersion",
+        "recorderMake",
+        "recorderModel",
+        "recorderSerialNumber",
+        "reelName",
+        "storageMediaSerialNumber",
+    };
+
+    for (const auto& label : nonEmptyAttribs) {
+        const ParamValue* found = spec.find_attribute(label,
+                                                      OIIO::TypeDesc::STRING);
+        auto value              = spec.get_string_attribute(label, "");
+
+        if (found && value.empty()) {
+            non_compliant_attr = label;
+            return false;
+        }
+    }
+
+    return true;
+}
+
+
+
+bool
+is_aces_container_compliant(const OIIO::ImageSpec& spec, std::string& reason)
+{
+    if (!is_spec_aces_container_channels_only(spec)) {
+        reason
+            = "Spec channel names do not match those required for an ACES Container.";
+        return false;
+    }
+
+    // Check data type
+    if (spec.format != OIIO::TypeDesc::HALF) {
+        reason
+            = "EXR data type is not 'HALF' as required for an ACES Container.";
+        return false;
+    }
+
+    // Check compression
+    std::string compression = spec.get_string_attribute("compression", "zip");
+    if (compression != "none") {
+        reason = "Compression is not 'none' as required for an ACES Container.";
+        return false;
```

Comment:
What's the best policy here?

For something like "doesn't have the right channels", the only thing to do is fail.

But for "wrong data type" or "wrong compression", we could force compliance by just *fixing it*.